mobile/src/board/BoardHooks.ts

```diff
@@ -19,18 +19,81 @@ export function useBoard(id: number | string): [Board | null, any | null] {
     }
   };
 
+  const handleMercureMessage = async ({ data }: { data: string }) => {
+    const actions: UserAction[] = JSON.parse(data);
+
+    for (const { kind, payload } of actions) {
+      const resolver = actionResolvers.get(kind);
+      if (resolver) {
+        await resolver(payload);
+      }
+    }
+  };
+
+  const actionResolvers = new Map<String, (payload: any) => Promise<void>>([
+    [
+      "ROTATE_REMAINING",
+      ({ direction }: { direction: string }) =>
+        new Promise((resolve) => {
+          setBoard((board) => {
+            if (!board) {
+              return null;
+            }
+            const remainingTile = board.state.remainingTile;
+
+            const newRotation =
+              direction == "CLOCKWISE"
+                ? remainingTile.rotation + 90
+                : remainingTile.rotation - 90;
+
+            return {
+              ...board,
+              state: {
+                ...board.state,
+                remainingTile: {
+                  tile: remainingTile.tile,
+                  rotation: newRotation,
+                },
+              },
+            };
+          });
+
+          setTimeout(resolve, 1000);
+        }),
+    ],
+    [
+      "GAME_STATE_CHANGE",
+      (gameState: number) =>
+        new Promise((resolve) => {
+          setBoard((board) => {
+            if (!board) {
+              return null;
+            }
+            return {
+              ...board,
+              state: {
+                ...board.state,
+                gameState,
+              },
+            };
+          });
+
+          setTimeout(resolve, 0);
+        }),
+    ],
+    ["PLACE_TILE", () => fetchBoard()],
```

Comment:
I think you could simply write : 
`["PLACE_TILE", fetchBoard],`and avoid redefining an anonymous proxy function
same for the next line